ui/components/multichain/network-list-menu/network-list-menu.js

```diff
@@ -352,6 +358,21 @@ export const NetworkListMenu = ({ onClose }) => {
                                     } else {
                                       dispatch(setActiveNetwork(network.id));
                                     }
+
+                                    // If presently on a dapp, communicate a change to
+                                    // the dapp via silent switchEthereumChain that the
+                                    // network has changed due to user action
+                                    if (
+                                      process.env.MULTICHAIN &&
+                                      useRequestQueue &&
+                                      selectedTabOrigin
+                                    ) {
+                                      setNetworkClientIdForDomain(
+                                        selectedTabOrigin,
+                                        network.id,
```

Comment:
don't love that this is property is called `id` -> should be `networkClientId` but I know you're not introducing that here.